samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs

```diff
@@ -1358,7 +862,71 @@ public virtual async Task<Response> GetDocumentsStatusAsync(Guid id, int? top =
         /// </code>
         /// 
         /// </remarks>
-        /// <param name="id"> Format - uuid.  The operation id. </param>
+#pragma warning disable AZC0002
+        public virtual AsyncPageable<BinaryData> GetTranslationsStatusAsync(int? top = null, int? skip = null, int? maxpagesize = null, IEnumerable<Guid> ids = null, IEnumerable<string> statuses = null, DateTimeOffset? createdDateTimeUtcStart = null, DateTimeOffset? createdDateTimeUtcEnd = null, IEnumerable<string> orderBy = null, RequestOptions options = null)
+#pragma warning restore AZC0002
+        {
+            options ??= new RequestOptions();
+            async Task<Page<BinaryData>> FirstPageFunc(int? pageSizeHint)
+            {
+                using var scope = _clientDiagnostics.CreateScope("DocumentTranslationClient.GetTranslationsStatus");
+                scope.Start();
+                try
+                {
+                    Response response = await RestClient.GetTranslationsStatusAsync(top, skip, maxpagesize, ids, statuses, createdDateTimeUtcStart, createdDateTimeUtcEnd, orderBy, options).ConfigureAwait(false);
+                    return LowLevelPagableHelpers.BuildPageForResponse(response, "value", "@nextLink");
+                }
+                catch (Exception e)
+                {
+                    scope.Failed(e);
+                    throw;
+                }
+            }
+
+            async Task<Page<BinaryData>> NextPageFunc(string nextLink, int? pageSizeHint)
+            {
+                using var scope = _clientDiagnostics.CreateScope("DocumentTranslationClient.GetTranslationsStatus");
+                scope.Start();
+                try
+                {
+                    Response response = await RestClient.GetTranslationsStatusNextPageAsync(nextLink, top, skip, maxpagesize, ids, statuses, createdDateTimeUtcStart, createdDateTimeUtcEnd, orderBy, options).ConfigureAwait(false);
```

Comment:
Multiple enumerations of `ids` in several async contexts. Potential deadlock.

Reply:
The Pageable abstraction does ensure only a single one of these async calls is ever running at once (it waits for `FirstPageFunc` to complete before calling `NextPageFunc` and ensures there are not multiple calls in flight.

Would you prefer that for any IEnumerable argument we force enumeration once (via calling .ToArray()) at the start of the method and just use that copy to the underlying service methods?

For example, here we could instead of something like:

```
options ??= new RequestOptions();
enumeratedIds = (ids)?.ToArray();
enumeratedStatueses = (statuses)?.ToArray();
enumeratedOrderBy = (orderBy)?.ToArray();
```

and then use the `enumerated` counterparts in the underlying calls to the RestClient methods.

Reply:
> Would you prefer that for any IEnumerable argument we force enumeration once (via calling .ToArray()) at the start of the method and just use that copy to the underlying service methods?

Yes!

Reply:
Pushed https://github.com/Azure/autorest.csharp/pull/1514/commits/1c8c4498ba04f4b6a0c285571172e131b7eac830 which does this (I used the suffix `Values` for the locals we create).